### ISSUE TITLE
Different column settings for orders and carts

### ISSUE THREAD

**Issue by lbw-dynamicweb** [dynamicweb/DynamicWeb#155]
In the Commerce module, viewing the Orders and Carts, it uses the same column setting. 
The relevant information for these 2 are very different, so it would be preferable to be able to set the setting per view.

**Comment by DynamicRas** [comment]
Yes, the reason is that Orders, Carts, Subscriptions, Quotes and Ledgers use the same list screen, so we would need to make separate ones. We will consider it, but not decided yet.

BR Rasmus Sanggaard

**Comment by nunoaguiar** [comment]
Hi,

That's a very interesting topic that I have thought about as well (separate tables for each object Rasmus mentions).

I know the data structure is very similar and having them be separate tables makes things a bit more "expensive" (performance wise at least - instead of updating some values in the same table, records need to be created and delete), but as over the years, as we see EcomOrders and EcomOrderlines grow, starts to become a bigger problem.

There could be a silver lining, which is to keep the Cart record when "upgrading to an Order", so the downgrade could adopt the original Cart Id (they would be tied to each other). At least from our end, it's always tricky to figure this stuff out, and ends up on the plate of a more seasoned developer because of it.

Another (odd) thing is with custom fields. Since they are all shared, the UI starts to become messy, specially looking at Quote and Ledgers, which typically have different fields than Carts/Orders.

I'm sure there's plenty of pros and cons. It's easy to speak about the cons currently, because we've been having a lot of exposure to them :) 

Best Regards,
Nuno Aguiar

**Comment by DynamicRas** [comment]
Hi,

Don't think I am / we are ready to open that door. But for this specific case I like to focus on the task at hand; making it possible to select unique columns for each list screen. :)

BR Rasmus Sanggaard

**Comment by nunoaguiar** [comment]
Fair enough Rasmus :) 
I know it would be a very big change.
